src/client.ts

```diff
@@ -94,6 +110,9 @@ export class Client {
       this.socket.connect(this.params.port, this.params.hostname)
     }
     if (params.socketTimeout) this.socket.setTimeout(params.socketTimeout)
+    this.hostname = params.hostname
+    this.leader = params.leader || false
```

Comment:
Useful tip: to define defaults, use ?? instead of ||. In this specific instance nothing changes, but if the default were `true` and params.leader was in fact false, `params.leader || true` evaluates as true while `params.leader ?? true` evaluates as false. Some linter settings even yell at you if you use || this way